bundles/org.smarthomej.binding.tr064/src/main/java/org/smarthomej/binding/tr064/internal/FritzboxActions.java

```diff
@@ -0,0 +1,134 @@
+/**
+ * Copyright (c) 2010-2021 Contributors to the openHAB project
+ * Copyright (c) 2021 Contributors to the SmartHome/J project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.smarthomej.binding.tr064.internal;
+
+import java.io.IOException;
+import java.nio.file.FileSystems;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeoutException;
+
+import javax.xml.soap.SOAPMessage;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.jetty.client.api.ContentResponse;
+import org.openhab.core.automation.annotation.RuleAction;
+import org.openhab.core.thing.binding.ThingActions;
+import org.openhab.core.thing.binding.ThingActionsScope;
+import org.openhab.core.thing.binding.ThingHandler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.smarthomej.binding.tr064.internal.dto.scpd.root.SCPDServiceType;
+import org.smarthomej.binding.tr064.internal.soap.SOAPRequest;
+import org.smarthomej.binding.tr064.internal.util.SCPDUtil;
+import org.smarthomej.binding.tr064.internal.util.Util;
+
+/**
+ * The {@link FritzboxActions} provides actions for managing scenes in groups
+ *
+ * @author Jan N. Klug - Initial contribution
+ */
+@ThingActionsScope(name = "tr065")
```

Comment:
I thought about something like that, too. But in fact the phonebook (which uses `tr064`) is also a Fritz!Box only action. Maybe it would be better to move that action to this file.

Reply:
This is also not allowed: https://github.com/openhab/openhab-addons/pull/8576#discussion_r611090749

Reply:
Okay. Then merge it I to the existing `ThingAction` class.